geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -216,3 +215,15 @@ class GeoMesaSparkKryoRegistrator extends KryoRegistrator {
     SimpleFeatureSerializers.simpleFeatureImpls.foreach(kryo.register(_, serializer, kryo.getNextRegistrationId))
   }
 }
+
+object GeoMesaSparkKryoRegistrator {
+  val typeCache: ConcurrentHashMap[String, SimpleFeatureType] = new ConcurrentHashMap[String, SimpleFeatureType]()
+
+  def putType(sft: SimpleFeatureType): Unit = {
+    GeoMesaSparkKryoRegistrator.typeCache.putIfAbsent(sft.getTypeName, sft)
```

Comment:
```putType``` can be called on the jvm on the executors too (where the typecache is likely empty), but I've only noticed it happen on the driver's jvm. Though I may have just forgotten to restart the kernel. I'll see if I can reproduce it tomorrow. But either way, I feel like sfts are final enough to just be put once